### PR TITLE
Rename Account and ClientData fixes #312

### LINKED ISSUE #312: Consider scoping Account and ClientData
They may be too generically named.

### PR DESCRIPTION
**Pull request by kpaulh**
Renames Account -> RelyingPartyAccount and ClientData -> AuthenticationClientData.

### CODE REVIEW

**Comment by equalsJeffH** [comment]
see also https://lists.w3.org/Archives/Public/public-webauthn/2016Dec/0171.html

**Comment by equalsJeffH** [comment]
[explicit link to #312]

**Comment by equalsJeffH** [comment]
> I'll defer to JeffH and others on this

LOL, ok.  here's what I'd choose if left to my own devices...

wrt `RelyingPartyAccount`:  `UserRelyingPartyAccount`

wrt `ClientData` nee `AuthenticationClientData`: well, there's a couple other associated "ClientData" terms  besides the latter, i.e.: `clientDataJSON` and `clientDataHash`.  

So I'd go with: `CollectedClientData`

**Comment by equalsJeffH** [comment]
I'd scrawled
> So I'd go with: `CollectedClientData`

Which implies, I spose, that we also perhaps should do this:

`clientDataJSON` -> `collectedClientDataJSON`

`clientDataHash` ->  `collectedClientDataHash`

**Comment by kpaulh** [comment]
I have no problems with that. `SourceClientData` seems fine, as well (and is slightly shorter).

Another suggestion for Account -> `RelyingPartyUserInfo`?

**Comment by vijaybh** [comment]
@equalsJeffH - do we really need to make the interface element names longer? They are already scoped by virtue of being in the interface. The type names are global and need to be more specific (and mostly don't matter to developers), but shouldn't these element names be shorter to avoid confusing developers?

**Comment by equalsJeffH** [comment]
@vijaybh pointed out
> shouldn't these element names be shorter to avoid confusing developers?

fine by me, was just thinking of naming consistency, note I qualified it with "should perhaps" ;-)

Perhaps the "clientdata" substring on its own is sufficient consistency.

@kpaulh suggested:
> Another suggestion for Account -> `RelyingPartyUserInfo`?

works for me.

**Comment by equalsJeffH** [comment]
@kpaulh @vijaybh 

Ok, so we are going with:
* `CollectedClientData`
* `RelyingPartyUserInfo`
and we are leaving `clientDataJSON` and `collectedClientDataHash` alone, yes?

**Comment by kpaulh** [comment]
Yep, sounds good to me, thanks!

**Comment by equalsJeffH** [comment]
Ok, I've approved these changes, it needs a merge-from-master-resolve-conflicts-and-re-review before merging, thanks.

File: index.bs [hunk 1]
```diff
@@ -479,7 +479,7 @@ When this method is invoked, the user agent MUST execute the following algorithm
 
     1. [=list/Append=] |result| to |clientExtensions|.
 
-1. Let |clientData| be a new {{ClientData}} instance whose fields are:
+1. Let |clientData| be a new {{CollectedClientData}} instance whose fields are:
```
Comment (equalsJeffH) [inline index.bs]: this is a nit, but it seems the WhatWG/W3C style is to say..
`Let |collectedClientData| be a new {{CollectedClientData}} instance...`

File: index.bs [hunk 2]
```diff
@@ -608,7 +608,7 @@ When this method is invoked, the user agent MUST execute the following algorithm
         algorithm returned an error, [=continue=].
     1. [=list/Append=] |result| to |clientExtensions|.
 
-1. Let |clientData| be a new {{ClientData}} instance whose fields are:
+1. Let |clientData| be a new {{CollectedClientData}} instance whose fields are:
```
Comment (equalsJeffH) [inline index.bs]: ditto

Comment (kpaulh) [inline index.bs]: Got it, making the change now. Thanks!

**Pull request merged**